### PR TITLE
Fixes Service Borg's RSF being unable to dispense items on floors

### PR DESCRIPTION
**Pull request by datlo**
The ability to dispense items on floors was removed by #8302 as an undocumented change. This fixes that.

🆑
fix: Fixed the ability of Service Borgs to spawn items on floors. Dosh!
/🆑

### CODE REVIEW

**Comment by FalseIncarnate** [comment]
Given the logic for that looks like a bad case of copy-paste (two cases that check for the same conditions), I think this would be a case where changing the functionality would be ideal.

In fact, prior to https://github.com/ParadiseSS13/Paradise/pull/8302 this was actually standard functionality (being able to spawn on floors), and was not documented as having been deliberately removed. Between this and the fact that it still does an initial check for tables and floors, I'd wager this is an excellent case to just make the RSF work on floors again.

Change line 64 to `else if(istype(A, /turf/simulated/floor))` and revert your change to the fail message

**Comment by datlo** [comment]
Did what you suggested, after reading through #8302 I believe you're correct. 

I look forward to our doshbot overlords

File: code/game/objects/items/weapons/RSF.dm [hunk 1]
```diff
@@ -61,7 +61,7 @@ RSF
 	var spawn_location
 	if(istype(A, /obj/structure/table))
 		spawn_location = A.loc
-	else if (istype(A, /obj/structure/table))
+	else if (istype(A, /turf/simulated/floor))
```
Comment (Crazylemon64) [inline code/game/objects/items/weapons/RSF.dm]: You could merge the table and floor checks with
```
var/turf/T = get_turf(A)
if(istype(T) && !T.density)
    spawn_location = T
```

**Comment by datlo** [comment]
Combined the two checks as suggested, I believe this is ready to be merged if there are no other issues

**Pull request merged**